### DISCUSSION TITLE
Memory usage help.

### DISCUSSION THREAD

**Discussion by lee-costa** [blakeblackshear/frigate#15743] [General]
A while back I [posted](https://github.com/blakeblackshear/frigate/discussions/15328) about some errors I was receiving after the latest update and no matter what I tried, I can't stop them from reappearing. At times, my Frigate instance is freezing and only a reboot fixes it. 

Today, it happened again and some research led me to checking my memory usage since the error was `Unable to keep up with recording segments in cache for backyard_cam. Keeping the 6 most recent segments out of 7 and discarding the rest...`. Now I am not entirely sure if this is even related but upon checking my memory usage, this is what I see:

```
              total        used        free      shared  buff/cache   available
Mem:          4.3Gi       4.0Gi       118Mi        85Mi       243Mi        69Mi
Swap:         2.0Gi       2.0Gi       0.0Ki
```
I am running two cameras only but with a fairly large resolution (1920 x 1080). I allocated 5GB of memory for this VM and don't understand why 5GB is not enough and swap is being used.

Is it possible that 5GB is not enough for two cameras? How come this never used to happen before when nothing has been changed on my end? The user on this [link](https://github.com/blakeblackshear/frigate/discussions/5392#discussioncomment-11377875) also states the same. 

Any help would be greatly appreciated.

**Comment by NickM-27** [comment]
your best bet would be to get a shell in the container and see what is using the memory
